subsys/bluetooth/host/gatt.c

```diff
@@ -84,8 +84,14 @@ static sys_slist_t callback_list;
 static sys_slist_t db;
 #endif /* CONFIG_BT_GATT_DYNAMIC_DB */
 
-static atomic_t init;
-static atomic_t service_init;
+enum gatt_global_flags {
+	GATT_INITIALIZED,
+	GATT_SERVICE_INITIALIZED,
+
+	GATT_NUM_FLAGS,
+};
+
+ATOMIC_DEFINE(gatt_flags, GATT_NUM_FLAGS);
```

Comment:
I think `static ATOMIC_DEFINE(...` would be more appropriate, since the variable is limited to this c-file. Otherwise looks good, thanks!